demo/api/src/db/fixtures/generators/document-generator.service.ts

```diff
@@ -0,0 +1,109 @@
+import { ExternalLinkBlock } from "@comet/blocks-api";
+import { File, PageTreeNodeInterface, PageTreeNodeVisibility, PageTreeService } from "@comet/cms-api";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { EntityRepository } from "@mikro-orm/postgresql";
+import { Inject, Injectable } from "@nestjs/common";
+import { LinkBlock } from "@src/common/blocks/linkBlock/link.block";
+import { Config } from "@src/config/config";
+import { CONFIG } from "@src/config/config.module";
+import { generateBlocksBlock } from "@src/db/fixtures/generators/blocks/blocks.generator";
+import { generateSeoBlock } from "@src/db/fixtures/generators/blocks/seo.generator";
+import { Link } from "@src/links/entities/link.entity";
+import { PageTreeNodeScope } from "@src/page-tree/dto/page-tree-node-scope";
+import { PageTreeNodeCategory } from "@src/page-tree/page-tree-node-category";
+import { Page } from "@src/pages/entities/page.entity";
+import faker from "faker";
+import slugify from "slugify";
+
+interface GeneratePageInput {
+    name: string;
+    scope: PageTreeNodeScope;
+    imageFiles: File[];
+    parentId?: string;
+}
+
+@Injectable()
+export class DocumentGeneratorService {
```

Comment:
This generator does more than. just generate documents. It also. creates the page tree nodes and attaches the document. Wouldn't it be better to split the functionality into multiple services?

Reply:
How exactly would you implement that?
I think the splitting up is a good, but I'd still prefer a simple usage of the "DocumentGenerator"

I'd expect it to work like this:
 - PageTreeNodeGenerator (handles attaching documents and so on)
 - DocumentGenerator (uses the PageTreeNodeGenerator but would work exactly the same as before, just without handling the code of the pageTreeNode itself but instead is calling the PageTreeNodeGenerator)

Reply:
I tried splitting it up but it seems unnecessary for me.

Instead of directly calling pageTreeService I'm now calling the pageTreeNode generator, I still need the same parameters: name, slug, attachedDocument, parentId, category and scope.

I would rather split it up like that:
PageGenerator
LinkGenerator
(and the generators do still create and attach the page tree nodes)

Reply:
Alright, fine for me 🙂